### PR TITLE
Ad e2e test retry

### PR DESCRIPTION
**Pull request by f2prateek**
Seems to fail with:
```
  10:25:44 as prateek on Thor-MBP in ~/dev/src/github.com/segmentio/analytics-node at e2e-test(!)
→ RUNSCOPE_TOKEN=3f58dc18-5990-41a6-9675-50be6e4e50c4 npm test

> analytics-node@3.1.1 test /Users/prateek/dev/src/github.com/segmentio/analytics-node
> standard && ava

/Users/prateek/dev/src/github.com/segmentio/analytics-node/node_modules/async-retry/lib/index.js:28
      let val
      ^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Module._extensions..js (module.js:416:10)
    at extensions.(anonymous function) (/Users/prateek/dev/src/github.com/segmentio/analytics-node/node_modules/ava/node_modules/require-precompiled/index.js:16:3)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/prateek/dev/src/github.com/segmentio/analytics-node/node_modules/ava/lib/process-adapter.js:100:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/prateek/dev/src/github.com/segmentio/analytics-node/test.js:12:1)
    at Module._compile (module.js:409:26)
    at extensions.(anonymous function) (/Users/prateek/dev/src/github.com/segmentio/analytics-node/node_modules/ava/node_modules/require-precompiled/index.js:13:11)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/prateek/dev/src/github.com/segmentio/analytics-node/node_modules/ava/lib/process-adapter.js:100:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/prateek/dev/src/github.com/segmentio/analytics-node/node_modules/ava/lib/test-worker.js:61:1)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
    at node.js:1043:3

  1 exception

  ✖ test.js exited with a non-zero exit code: 1
npm ERR! Test failed.  See above for more details.

```

### CODE REVIEW

**Comment by f2prateek** [comment]
@vadimdemedes tried using async-retry as you suggested in https://github.com/segmentio/analytics-node/pull/134#issuecomment-344623068, but seeing this error https://github.com/segmentio/analytics-node/pull/135#issue-274258309

File: test.js [hunk 1]
```diff
@@ -452,7 +453,7 @@ test('alias - require previousId and userId', t => {
   })
 })
 
-test('e2e', async t => {
+test('e2e', retry(async t => {
```
Comment (stephenmathieson) [inline test.js]: Because we call `t.fail()`, `async-retry` won't work here. Instead, I think we need to throw an error.

**Comment by vadimdemedes** [comment]
@f2prateek what Node.js version are you on?

**Comment by f2prateek** [comment]
@vadimdemedes `→ node --version v4.8.5`

**Comment by vadimdemedes** [comment]
Weird, should be fine. Try upgrading to Node.js 8 and reinstalling all the dependencies just to be sure.

**Pull request closed without merging**